coriolis_openstack_utils/actions.py

```diff
@@ -935,3 +938,149 @@ def execute_operations(self):
+                return {"done": True, "result": subnet['id']}
+
+        if conflicting:
+            raise Exception("Found multiple destination subnets with "
+                            "with name %s! Aborting subnet "
+                            "migration." % dest_subnet_name)
+
+        return {"done": False, "result": None}
+
+    def equivalent_to(self, other_action):
+        if other_action.action_type == self.action_type:
+            if self.payload['source_name'] == (
+                    other_action.payload.get('source_name')):
+
+                src_network_id = self.payload['src_network_id']
+                src_other_network_id = (
+                    other_action.payload.get('src_network_id'))
+
+                if src_network_id == src_other_network_id:
+                    dest_network_id = self.payload['dest_network_id']
+                    dest_other_network_id = other_action.payload.get(
+                        'dest_network_id')
+
+                    return dest_network_id == dest_other_network_id
+        return False
+
+    def print_operations(self):
+        super(SubnetCreationAction, self).print_operations()
+        subnet_name = self.get_new_subnet_name()
+        LOG.info(
+            "Create new destination subnet named '%s'." % subnet_name)
+
+    def get_new_subnet_name(self):
+        return self.subnet_name_format % {
+            "original": self.payload["source_name"]}
+
+    def get_source_tenant_id(self):
+        src_subnet_list = subnets.list_subnets(
+            self._source_client, self.payload['src_network_id'],
+            self.payload['source_name'])
+
+        if not src_subnet_list:
+            raise Exception("Source Subnet %s in network %s not found!"
+                            % (self.payload['source_name'],
+                               self.payload['src_network_id']))
+
+        src_subnet = src_subnet_list[0]
+
+        return (src_subnet.get('tenant_id') or
+                src_subnet.get('project_id'))
+
+    def get_destination_tenant_id(self):
+        dest_network = networks.get_network(
+            self._destination_client, self.payload['dest_network_id'])
+
+        return (dest_network.get('tenant_id') or
+                dest_network.get('project_id'))
+
+    def create_subnet_body(self, description):
+        src_subnet_name = self.payload['source_name']
+        src_tenant_id = self.get_source_tenant_id()
+        dest_tenant_id = self.get_destination_tenant_id()
+        dest_subnet_name = self.get_new_subnet_name()
+        dest_network_id = self.payload['dest_network_id']
+        src_body = subnets.get_body(
+            self._source_client, src_tenant_id, src_subnet_name)
+        body = {'name': dest_subnet_name,
+                'tenant_id': dest_tenant_id,
+                'project_id': dest_tenant_id,
+                'description': description,
+                'network_id': dest_network_id}
+
+        body.update(src_body)
```

Comment:
Pleas eonly set the keys from the `src_body` if they are *not* present in the `body` already